File: node/hello.js
const { MongoClient } = require("mongodb");
const { version } = require("mongodb/package.json");

async function main() {
  console.log(`Ping from mongodb ${version} ...`);
  const client = new MongoClient("mongodb://localhost:27017");
  await client.connect();
  await client.db("admin").command({ ping: 1 });
  await client.close();
  console.log(`Ping from mongodb ${version} ... OK`);
}

main().catch(console.error);

File: go/hello.go
package main

import (
	"context"
	"fmt"
	"log"
	"runtime/debug"

	"go.mongodb.org/mongo-driver/v2/bson"
	"go.mongodb.org/mongo-driver/v2/mongo"
	"go.mongodb.org/mongo-driver/v2/mongo/options"
)

func driverVersion() string {
	info, ok := debug.ReadBuildInfo()
	if !ok {
		return "unknown"
	}
	for _, dep := range info.Deps {
		if dep.Path == "go.mongodb.org/mongo-driver/v2" {
			return dep.Version
		}
	}
	return "unknown"
}

func main() {
	version := driverVersion()
	fmt.Printf("Ping from mongo-go-driver %s ...\n", version)

	client, err := mongo.Connect(options.Client().ApplyURI("mongodb://localhost:27017").SetMonitor(makeMonitor()))
	if err != nil {
		log.Fatal(err)
	}
	defer client.Disconnect(context.Background())

	err = client.Database("admin").RunCommand(context.Background(), bson.D{{Key: "ping", Value: 1}}).Err()
	if err != nil {
		log.Fatal(err)
	}

	fmt.Printf("Ping from mongo-go-driver %s ... OK\n", version)

	// TODO: add your code here
}
